servers/tests/org.fusesource.ide.server.karaf.core.tests/src-test/org/fusesource/ide/server/karaf/core/server/subsystems/KarafUtilsTest.java

```diff
@@ -0,0 +1,126 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.fusesource.ide.server.karaf.core.server.subsystems;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.eclipse.core.runtime.CoreException;
+import org.fusesource.ide.foundation.core.util.IOUtils;
+import org.fusesource.ide.foundation.core.util.Strings;
+import org.fusesource.ide.server.karaf.core.util.KarafUtils;
+import org.junit.Test;
+
+/**
+ * @author lhein
+ */
+public class KarafUtilsTest {
+	
+	private static final String MANIFEST_BLURP = "Manifest-Version: 1.0\n" + 
+			"Bundle-ManifestVersion: 2\n" + 
+			"Bundle-Name: Test Bundle\n" + 
+			"Bundle-SymbolicName: com.sample.mytest.BundleName\n";
+	
+	@Test
+	public void testGetBundleVersionFromManifest() throws CoreException, IOException {
+		File mfFile = createTempManifest("1.0.0");
+		String version = KarafUtils.getBundleVersionFromManifest(mfFile);
+		assertEquals("1.0.0", version);
+		
+		mfFile = createTempManifest("1.0.0.SNAPSHOT");
```

Comment:
it is better to split tests to follow the Given/Then/When pattern one time per method.
there is several advantages for that:
- you will see every specific that are failing (instead of the first one in the method)
- it forces to give a new more specific name for each test methiod which helps to undertand what is tested and what is failing in test report
- shorter methods